src/libs/actions/NetworkRequestQueue.js

```diff
@@ -21,10 +21,19 @@ function saveRetryableRequests(retryableRequests) {
     Onyx.set(ONYXKEYS.NETWORK_REQUEST_QUEUE, persistedRequests);
 }
 
-function removeRetryableRequest(request) {
+function removeRetryableRequest(request, error) {
     retryMap.delete(request);
     persistedRequests = _.reject(persistedRequests, r => _.isEqual(r, request));
     Onyx.set(ONYXKEYS.NETWORK_REQUEST_QUEUE, persistedRequests);
+
+    if (error && request.data) {
+        const {reportID, clientID: optimisticReportActionID} = request.data;
+        if (reportID && optimisticReportActionID) {
+            Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`, {
+                [optimisticReportActionID]: null,
+            });
+        }
+    }
```

Comment:
Okay. But there will be a case if the file upload is done during offline and then we are back online even though the request fails it will keep on showing loading as there currently is no mechanism for removing the optimisticReport from persisted request.

Reply:
We are working on persisted requests and a retry mechanism. This is not related to this issue and should not be handled in this PR. If there is some issue different from this one, please report it on the Slack channel. 

lastly, NetworkQueue logic is not related to reports in any way so no code should modify reports and other entities from these handlers. If there are side-effects that need to be done on request completion/failure then they have a dedicated place for the purpose. An action should handle it and component|background task should control it.